doc/user-guide/data_model.rst

```diff
@@ -0,0 +1,251 @@
+representation of the data.  For example:
+
+.. jupyter-execute::
+
+   >>> import pyvista
+   >>> wrapped = pyvista.wrap(vtk_array)
+   >>> wrapped
+
+Note that when wrapping the underlying VTK array, we actually perform
+a shallow copy of the data.  In other words, we pass the pointer from
+the underlying C array to the numpy :class:`numpy.ndarray`, meaning
+that the two arrays are now efficiently linked.  This means that we
+can change the array using numpy array indexing and have it modified
+on the "VTK side".
+
+.. jupyter-execute::
+
+   >>> wrapped[0, 0] = 10
+   >>> vtk_array.GetValue(0)
+
+Or we can change the value from the VTK array and see it reflected in
+the numpy wrapped array.  Let's change the value back:
+
+.. jupyter-execute::
+
+   >>> vtk_array.SetValue(0, 0)
+   >>> wrapped[0, 0]
+
+
+Using a Python List
+~~~~~~~~~~~~~~~~~~~
+PyVista supports the use of Python lists, and you could define a your
+points using a nested list of lists via:
+
+.. jupyter-execute::
+
+   >>> points = [[0, 0, 0],
+   ...           [1, 0, 0],
+   ...           [0.5, 0.667, 0]]
+
+When used in the context of :class:`pyvista.PolyData` to create the
+mesh, this list will automatically be wrapped using numpy and then
+passed to VTK.  This avoids any looping overhead and while still
+allowing you to use native python classes.
+
+Finally, let's show how we can use these three objects in the context
+of a PyVista geometry class.  Here, we create a simple point mesh
+containing just the three points:
+
+.. jupyter-execute::
+   
+   >>> from_vtk = pyvista.PolyData(vtk_array)
+   >>> from_np = pyvista.PolyData(np_points)
+   >>> from_list = pyvista.PolyData(points)
+
+These point meshes all contain three points and are effecively
+identical.  Let's show this by accessing the underlying points array
+from the mesh, which is represented as a :class:`pyvista_ndarray`
+
+.. jupyter-execute::
+
+   >>> from_vtk.points
+
+And show that these are all identical
+
+.. jupyter-execute::
+
+   >>> assert np.allclose(from_vtk.points, from_np.points)
+   >>> assert np.allclose(from_vtk.points, from_list.points)
+   >>> assert np.allclose(from_np.points, from_list.points)
+
+Finally, let's plot this (very) simple example using PyVista's
+:func:`pyvista.plot` method.  Let's make this a full example so you
+can see the entire process.
+
+.. pyvista-plot::
+   :context:
+
+   >>> import pyvista
+   >>> points = [[0, 0, 0],
+   ...           [1, 0, 0],
+   ...           [0.5, 0.667, 0]]
+   >>> mesh = pyvista.PolyData(points)
+   >>> mesh.plot(show_bounds=True, cpos='xy', point_size=20)
+
+We'll get into PyVista's data classes and attributes later, but for
+now we've show how create a simple mesh containing only points.  To
+create a surface, we must specify the connectivity of the geometry, and
+to do that we need to specify the cells (or faces) of this mesh.
+
+
+Geometry and Mesh Connectivity within PyVista
```

Comment:
I would write "mesh connectivity / topology", since topology is used in other docs and in mathematical contexts, so it's good to know both terms.